File: code2040_step2_aburt.py
import requests

# API access token
myToken = '3624f3214a21f60da7488a0716e6eee3'

# Code2040 API POST request for challenge info
step2 = requests.post('http://challenge.code2040.org/api/reverse', data={"token": myToken})
step2RequestResponse = step2.text
print(step2RequestResponse)

# Reverses API string using extended slices
step2RequestResponseReversed = step2RequestResponse[::-1]
print(step2RequestResponseReversed)
# Dictionary Created to send with POST request validation
step2dataToPost = {
    'string': step2RequestResponseReversed,
    'token': myToken
}
# Sends API POST request for challenge validation and displays text response
step2Validate = requests.post('http://challenge.code2040.org/api/reverse/validate', json=step2dataToPost)
print(step2Validate.text)

File: code2040_step2_aburt.js
var Promise = require("bluebird");
var http = require("http");
http.post = require("http-post");
var url = "http://challenge.code2040.org/api/reverse";
var myAPIToken = '3624f3214a21f60da7488a0716e6eee3';
var validateUrl = 'http://challenge.code2040.org/api/reverse/validate';
var infoToken= {
	'token': myAPIToken
}

function result(){ 
	return new Promise(function (resolve){
	var myData = "";
	var requestResponse = http.post(url,infoToken, function(response){
	response.on('data', function(data) {
		myData = data;
		//console.log(myData);
		resolve (data);
	}).setEncoding('utf8');
	response.on("error",function(error){
		console.log(error);
	})
	
    });


})

};

result().then(
function(data){
	console.log(data);
	var myData = data.split('').reverse().join('');
	console.log(myData);
	//console.log(myData);
	
var dataToPost = {
	'string' : myData,
	'token' : myAPIToken
}

var challengeValidate = http.post(validateUrl,dataToPost,function(response){
	response.on("data",function(data){
		console.log(data);
	}).setEncoding("utf8");


})
});

//reverse();
